pkg/resolver/resource/resolver.go

```diff
@@ -137,3 +137,49 @@ func (c *Resolver) getHostMetaDocumentFromEndpoint(hostMetaEndpoint string) (dis
 
 	return hostMetaDocument, nil
 }
+
+// WebFingerResolver is used for resolving WebFinger resources.
+type WebFingerResolver struct {
+	httpClient *http.Client
+}
+
+// NewWebFingerResolver returns a new WebFingerResolver.
+func NewWebFingerResolver(httpClient *http.Client) WebFingerResolver {
+	return WebFingerResolver{httpClient: httpClient}
+}
+
+// ResolveWebFingerResource attempts to resolve the given WebFinger resource from domainWithScheme.
+func (w *WebFingerResolver) ResolveWebFingerResource(domainWithScheme, resource string) (discoveryrest.JRD, error) {
```

Comment:
This code either belongs to webfinger/client or we should move the code from webfinger/client here. We can discuss this. No need for two objects doing the same functionality.